homeassistant/components/heos/config_flow.py

```diff
@@ -100,3 +116,72 @@ async def async_step_reconfigure(
             data_schema=vol.Schema({vol.Required(CONF_HOST, default=host): str}),
             errors=errors,
         )
+
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_USERNAME): selector.TextSelector(),
+        vol.Optional(CONF_PASSWORD): selector.TextSelector(
+            selector.TextSelectorConfig(type=selector.TextSelectorType.PASSWORD)
+        ),
+    }
+)
+
+
+class HeosOptionsFlowHandler(OptionsFlow):
+    """Define HEOS options flow."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            authentication = CONF_USERNAME in user_input or CONF_PASSWORD in user_input
+            if authentication and CONF_USERNAME not in user_input:
+                errors[CONF_USERNAME] = "username_missing"
+            if authentication and CONF_PASSWORD not in user_input:
+                errors[CONF_PASSWORD] = "password_missing"
+
+            if not errors:
+                heos = cast(
+                    Heos, self.config_entry.runtime_data.controller_manager.controller
+                )
+
+                try:
+                    if user_input:
+                        # Attempt to login
+                        try:
+                            await heos.sign_in(
+                                user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+                            )
+                            _LOGGER.info(
+                                "Successfully signed-in to HEOS Account: %s",
+                                heos.signed_in_username,
+                            )
+                        except CommandFailedError as err:
+                            if err.error_id in (6, 8, 10):  # Auth-specific errors
+                                errors["base"] = "invalid_auth"
+                                _LOGGER.info(
+                                    "Failed to sign-in to HEOS Account: %s", err
+                                )
+                            else:
+                                raise  # Re-raise unexpected error
+                    else:
+                        # Log out
+                        await heos.sign_out()
```

Comment:
I'm not a big fan of the surrounding `try` (but that could be personal preferences), because imo it's harder to read